File: cpp/3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int f[128], ret=0,i=0,j=0;
        memset(f,-1,sizeof(f));
        while(j<s.size()){
            if(f[s[j]]<i)ret=max(j-i+1,ret);
            else i=f[s[j]]+1;
            f[s[j]]=j++;
        }
        return ret;
    }
};

File: py/3.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        i = j = ret = 0
        m = {}
        while j < len(s) :
            if m.get(s[j], -1) < i :
                ret = max(ret, j - i + 1)
            else :
                i = m[s[j]] + 1
            m[s[j]] = j
            j += 1
        return ret
